src/main/java/core/basesyntax/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,207 @@
+            carUpdateStatement.setLong(2, car.getManufacturer().getId());
+            carUpdateStatement.setString(3, car.getModel());
+            carUpdateStatement.setLong(4, car.getId());
+            carUpdateStatement.executeUpdate();
+            deleteLinksStatement.setLong(1,car.getId());
+            deleteLinksStatement.executeUpdate();
+            newLinksStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                newLinksStatement.setLong(2, driver.getId());
+                newLinksStatement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to UPDATE car in DB: "
+                    + car, e);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteString = "UPDATE cars SET deleted=true WHERE car_id=?;";
+        int result;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(deleteString)) {
+            statement.setLong(1, id);
+            result = statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to DELETE car from DB, id=" + id, e);
+        }
+        return result > 0;
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getCarsByDriverString = "SELECT car_id FROM cars_drivers WHERE driver_id=?;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarsStatement = connection
+                        .prepareStatement(getCarsByDriverString)) {
+            getCarsStatement.setLong(1, driverId);
+            ResultSet carsResultSet = getCarsStatement.executeQuery();
+            while (carsResultSet.next()) {
+                cars.add(get(carsResultSet.getLong("car_id")).get());
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to GET cars by driver_id from DB, id="
+                    + driverId, e);
+        }
+        return cars;
+    }
+
+    private Car makeCar(ResultSet resultSet) {
+        try {
+            String manufacturerName = resultSet.getString("made_by");
+            String manufacturerCountry = resultSet.getString("country");
+            Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+            String carModel = resultSet.getString("model");
+            Long carId = resultSet.getObject("car_id", Long.class);
+            Manufacturer manufacturer = new Manufacturer(manufacturerName, manufacturerCountry);
+            manufacturer.setId(manufacturerId);
+            Car car = new Car(carModel, manufacturer);
+            car.setId(carId);
+            return car;
+        } catch (SQLException e) {
```

Comment:
we already catching this exception in the upper method(in each method where you are using makeCar())

Reply:
done